shaq_overflow/app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,58 @@
+class VotesController < ApplicationController
+
+  def create
+    if params[:votableType] == "Answer"
+      answer_votes
+    elsif params[:votableType] == "Question"
+      question_votes
+    elsif params[:votableType] == "Comment"
+      comment_votes
+    end
+  end
+
+  def answer_votes
+  if request.xhr?
+      if params[:direction] == "up"
+        vote = Vote.create(votable_type: params[:votableType], user_id: current_user.id, value: 1, votable_id: params[:answerId])
+        @answer = Answer.find(params[:answerId])
+      else
+        vote = Vote.create(votable_type: params[:votableType], user_id: current_user.id, value: -1, votable_id: params[:answerId])
+        @answer = Answer.find(params[:answerId])
+      end
+        x = @answer.net
+        x.to_json
+        render json: x
+    end
+  end
+
+  def question_votes
+    if request.xhr?
+      if params[:direction] == "up"
+        vote = Vote.create(votable_type: params[:votableType], user_id: current_user.id, value: 1, votable_id: params[:questionId])
+        @question = Question.find(params[:questionId])
+      else
+        vote = Vote.create(votable_type: params[:votableType], user_id: current_user.id, value: -1, votable_id: params[:questionId])
+        @question = Question.find(params[:questionId])
+      end
+        x = @question.q_net
+        x.to_json
+        render json: x
+    end
+  end
+
+  def comment_votes
+    if request.xhr?
+      if params[:direction] == "up"
+        vote = Vote.create(votable_type: params[:votableType], user_id: current_user.id, value: 1, votable_id: params[:commentId])
+        @comment = Comment.find(params[:commentId])
+      else
+        vote = Vote.create(votable_type: params[:votableType], user_id: current_user.id, value: -1, votable_id: params[:commentId])
+        @comment = Comment.find(params[:commentId])
+      end
+        x = @comment.c_net
+        x.to_json
+        render json: x
+    end
```

Comment:
I feel like these three methods could be collapsed into one. You are already passing votable_type as a parameter but then you pass one of comment_id, answer_id, question_id - so needing these three methods. If you just passed the parameter as votable_id, you'd only need one. Remember the DRY principle and find a way to refactor thiese to one method. Similarly there's no real reason to have c_net, q_net and net have different names.